### DISCUSSION TITLE
Upon serialization some fields are missing

### DISCUSSION THREAD

**Discussion by Zombobot1** [typegoose/typegoose#547] [Q&A] [answered]
## Versions

- System: windows
- NodeJS: 16.0.0
- Typescript: 4.2.4
- Compiler / Transpiler: ts-node 
- Typegoose(NPM): 8.0.0-beta.13
- mongoose: 5.11.19
- mongodb: 4.4.4

## What is the Problem?

Upon serialization some fields are missing. In the example bellow the __price__  attribute disappears. 

## Code Example

```ts
class Toy {
  @prop()
  name?: string;
}

class ExpensiveToy extends Toy {
  @prop()
  price?: number;
}

export class Kitten {
  @prop()
  name?: string;

  @prop({ type: () => Toy || ExpensiveToy })
  toy?: Array<Toy | ExpensiveToy>;
}
export const KittenModel = getModelForClass(Kitten);

async () => {
  const kitten: Kitten = { name: 'fluffy', toy: [{ name: 'ball', price: 1 }] };
  const { _id: kittenId } = await KittenModel.create(kitten);
  const fluffy = await KittenModel.findOne({ name: 'fluffy' }).exec();
  console.log(fluffy); //  toy: [ { _id: 60b931c09849915e68c19eb8, name: 'ball' } ]
}
```

## Do you know *why* it happens?

In this case if I replace type to `() => ExpensiveToy || Toy` the problem is resolved. However if I add another toy class with a new attribute it will disappear.

**Comment by hasezoey** [comment, marked as answer]
Sorry to say this, but `type: () => Toy || ExpensiveToy` is not valid, this will return an boolean (and should be always `true`)

i think you may want to look into [nested-discriminators](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#discriminators)
